src/odemis/gui/comp/overlay/base.py

```diff
@@ -241,7 +241,10 @@ def draw(self, ctx, canvas_padding=None, view_width=None, view_height=None):
         if self.background:
             margin_x = 6  # margins for better representation of background
             margin_y = 10
-            ctx.set_source_rgba(*(self.background + (self.opacity,)))
+            if len(self.colour) == 4:
+                ctx.set_source_rgba(*self.background[:-1], self.background[-1] * self.opacity)
+            else:
+                ctx.set_source_rgba(*self.background)
```

Comment:
Next to the else, you can add a comment mentioning that self.background should be len of 3.
In such case, you should use `*(self.background + (self.opacity,)`

Reply:
I was thinking about this, but came to the conclusion that self.opacity is also used for the letters right? So do we want to have both foreground and background being multiplied by the same opacity value (if not specified as fourth value)?

Reply:
I think that if colour or background is only 3 values it should be interpreted as the alpha==1. Right now your code is going to fail as it passes 3 values to a function that expects 4.

As to how to use .opacity when mixed with colour/background having already an alpha channel: it's a little up to us. I think that right now it's not doing anything properly. For the colour, it uses opacity if the colour is 3 channels, and don't use it if it's 4 channels. For the background and shadow it's used always multiplied by 0.7. So it's not consistent.

An alternative would be to drop entirely .opacity, which actually doesn't seem every used. I would be in favour of this last option. Then the user could either set colours with 3 values, in which case opacity is 1, or 4 values, in which case the opacity is the alpha channel.

If we go this way, you could just keep your current code simple (eg, use self.background or self.background + 1). Later on we would create another PR to drop the .opacity attribute.

Reply:
Yes dropping the .opacity seems to make most sense.